### PR TITLE
Deployment preparation: (capped) `SingleRecipientGauge`

### LINKED ISSUE #2217: Create Single Recipient (Stakeless) Capped Gauge
We have stakeless gauges, and capped gauges: but no stakeless capped gauges.

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `SingleRecipientGauge`

Build-info: master@https://github.com/balancer-labs/balancer-v2-monorepo/commit/e670661a41c7bc44dd147548c24774eff8e18f61

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts and expected output folders 
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Deprecating old tasks checklist: 

- [x] Old task was moved to `deprecated/` directory
- [x] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
- [x] Old task imports in `index.ts`, `input.ts` and fork tests are updated
- [x] Old task `README` is updated with a warning sign, a link to the replacement task and a short description 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

Closes #2217

### CODE REVIEW

File: pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/readme.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# 2023-02-15 - Single Recipient Gauge Factory V2
+
+Deployment of the `SingleRecipientGaugeFactory`, for liquidity gauges which forward their entire BAL allowance onto a single address.
+This version supersedes [single recipient gauges V1](../deprecated/20220325-single-recipient-gauge-factory/), featuring weight-capped gauges.
```
Comment (EndymionJkb) [inline pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/readme.md:4]:
```suggestion
This version adds a weight cap to single recipient gauges, and supersedes [single recipient gauge factory V1](../deprecated/20220325-single-recipient-gauge-factory/).
```

**Review by nventuro — APPROVED** [review]
Looks good!

However, don't deploy yet. There's a last minute small feature we want to add to the gauges.

File: pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/test/task.fork.ts [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      },
+      { value: ethToJoin }
+    );
+
+    await bal80weth20Pool.connect(balWhale).approve(veBAL.address, MAX_UINT256);
+    const currentTime = await currentTimestamp();
+    await veBAL
+      .connect(balWhale)
+      .create_lock(await bal80weth20Pool.balanceOf(balWhale.address), currentTime.add(MONTH * 12));
+  });
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/test/task.fork.ts:116]: My fault  :) But it is also painful to find an appropriate veBAL holder, and it constrains the block. So pick your poison I guess, but to me this is straightforward (and documents the process)

[the remark this answers is not in the document]
Comment (jubeira) [reply pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/test/task.fork.ts:116]: I have to agree with Jeff.

As discussed offline, the problem is not actually finding _a_ veBAL holder, but to find a _big one_ with available voting power at an arbitrary block number.

I first tried impersonating one but I ran into a couple of issues trying to get past the cap (even lowering it down). In the end, this is repeatable for future tests and gives you more flexibility and control over the amount of veBAL available for testing.

[the remark this answers is not in the document]
Comment (nventuro) [reply pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/test/task.fork.ts:116]: _fine_ :sob:

**Comment by jubeira** [comment]
New features: #2267.

This will impact the commit where the build info is generated from.

**Pull request merged**